packages/kbn-i18n/src/loader.ts

```diff
@@ -137,9 +137,13 @@ export async function getTranslationsByLocale(locale: string): Promise<PlainMess
 
   return files.length
     ? files.reduce(
-        (messages, file) => ({
+        (messages: { locale: string; messages?: Translation['messages'] }, file) => ({
           ...messages,
           ...loadedFiles[file],
+          messages: {
+            ...(loadedFiles[file].messages || {}),
+            ...(messages.messages || {}),
+          },
         }),
         { locale }
       )
```

Comment:
__nit:__ on the next line, if we make `messages` required, then we'll be returning `{ locale, messages: {} }` instead of `{}`.